src/main/java/data_access/grade_api/outgoing_data_formatting/CollectionJSONBuilder.java

```diff
@@ -54,21 +44,25 @@ private JSONObject buildMediaCollection(MediaCollection<
 
         collectionJSON.put("name", mediaCollection.getName());
         collectionJSON.put("collectionType", mediaCollection.getCollectionType());
-        collectionJSON.put("mediaType", mediaCollection.getMediaType().getName());
-        final JSONArray mediaJSON = new JSONArray();
-        collectionJSON.put("media", mediaJSON);
-
+        final String mediaType = mediaCollection.getMediaType().getName();
+        collectionJSON.put("mediaType", mediaType);
+        final JSONArray mediaJSONArray = new JSONArray();
         for (AbstractMedia media : mediaCollection) {
-            final AbstractMediaJSONBuilder builder = builders.get(
-                    mediaCollection.getMediaType().getName());
-            if (builder != null) {
-                mediaJSON.put(builder.buildJSON(media));
-            }
-            else {
-                throw new UnsupportedOperationException("Unsupported media type: "
-                        + mediaCollection.getMediaType().getName());
-            }
+            mediaJSONArray.put(buildMedia(media));
         }
+        collectionJSON.put("media", mediaJSONArray);
         return collectionJSON;
     }
+
+    private JSONObject buildMedia(AbstractMedia media) {
+        final JSONObject mediaJSON;
+        final MediaJSONBuilder builder = new MediaJSONBuilder();
+        if (media instanceof Movie) {
```

Comment:
Could you explain why you've eliminated the builder-media mapping in favor of an if statement? Doesn't that violate OCP?